src/main/java/seedu/agendum/storage/XmlToDoListStorage.java

```diff
@@ -70,4 +71,10 @@ public void saveToDoList(ReadOnlyToDoList toDoList, String filePath) throws IOEx
     public void saveToDoList(ReadOnlyToDoList toDoList) throws IOException {
         saveToDoList(toDoList, filePath);
     }
+
+    @Override
+    public void setToDoListFilePath(String filePath) {
+        assert StringUtil.isValidFilePath(filePath);        
```

Comment:
Did you delete the old storage file with FileUtil.deleteFileAtPath(this.filePath)?
My old files are not deleted >.<